python/unblob/handlers/archive/ar.py

```diff
@@ -15,6 +17,56 @@
 SIGNATURE_LENGTH = 0x8
 
 
+class Seekable:
+    """Adapter for file_utils.Seekable.
+
+    Changes the parameter names, as they are different for arpy and unblob.File.
+    """
+
+    def __init__(self, arpy_file: arpy.ArchiveFileData):
+        self._arpy_file = arpy_file
+
+    def read(self, n: Optional[int] = None) -> bytes:
+        return self._arpy_file.read(n)
+
+    def seek(self, pos: int, whence: int = io.SEEK_SET) -> int:
+        return self._arpy_file.seek(pos, whence)
+
+
+class ArExtractor(Extractor):
+    def extract(self, inpath: Path, outdir: Path) -> Optional[ExtractResult]:
+        fs = FileSystem(outdir)
+
+        with arpy.Archive(inpath.as_posix()) as archive:
+            archive.read_all_headers()
+
+            for name in sorted(archive.archived_files):
+                archived_file = archive.archived_files[name]
+
+                try:
+                    path = Path(name.decode())
+                except UnicodeDecodeError:
```

Comment:
I would copy [integration test files from arpy](https://github.com/viraptor/arpy/tree/master/test) into `tests/integration/archive/ar/__input__` to make sure we're covering these exceptions (and maybe others). They're just files, not code, so I don't see any licensing issue.